mediathread/templates/projects/composition_assignment.html

```diff
@@ -120,22 +120,32 @@ <h1 class="page-title">
                         <div class="dropdown d-inline ml-1 response-dropdown">
                             <button class="btn btn-primary dropdown-toggle" type="button" id="student-response-dropdown"
                                 data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                                {{responses|length}} of {{students|length}} students responded
+                                {% get_submitted responses as submittedResponses %}
+                                {{submittedResponses|length}} of {{students|length}} Students Responded
                             </button>
                             <div class="dropdown-menu dropdown-menu-right" aria-labelledby="student-response-dropdown">
-                                {% for student in students %}
-                                    {% student_response responses student as response %}
-                                    {% if response.date_submitted %}
-                                        <a href="{% url 'project-workspace' request.course.id response.id %}"
+                                <h6 class="dropdown-header">Submitted</h6>
+                                {% for submitted in submittedResponses %}
+                                    {% feedback submitted as feedback %}
+                                        <a href="{% url 'project-workspace' request.course.id submitted.id %}"
                                             class="dropdown-item" type="button">
-                                            {% public_name for student %} ({{response.date_submitted|date:"m/d/Y"}})
+                                            {{submitted.attribution}}
```

Comment:
Any reason not to use "public_name" here as you do in the loop below?

Reply:
I think you can only derive public_name from "students" or user data and since I was iterating through responses I had to use attribution to get the authors.

Reply:
Ah okay, I get it.

Great work on this. I'll merge now!